js/views/userPage/Store.js

```diff
@@ -0,0 +1,404 @@
+import BaseVw from '../baseVw';
+import ListingDetail from '../modals/listingDetail/Listing';
+import StoreListings from './StoreListings';
+import CategoryFilter from './CategoryFilter';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    if (!this.collection) {
+      throw new Error('Please provide a collection.');
+    }
+
+    if (!this.model) {
+      throw new Error('Please provide a model.');
+    }
+
+    this.countryList = getTranslatedCountries(app.settings.get('language'));
+
+    this.filter = {
+      category: 'all',
+      searchTerm: '',
+      sortBy: 'PRICE_ASC',
+      freeShipping: false,
+    };
+
+    this.listenTo(this.collection, 'request', this.onRequest);
+    this.listenTo(this.collection, 'update', this.onUpdateCollection);
+
+    if (options.initialFetch) {
+      this.fetch = options.initialFetch;
+      this.onRequest(this.collection, this.fetch);
+    }
+  }
+
+  className() {
+    return 'userPageStore';
+  }
+
+  events() {
+    return {
+      'click .js-retryFetch': 'onClickRetryFetch',
+      'change .js-filterFreeShipping': 'onFilterFreeShippingChange',
+      'change .js-shipsToSelect': 'onShipsToSelectChange',
+      'change .js-filterShipsTo': 'onShipsToCheckBoxChange',
+      'keyup .js-searchInput': 'onKeyupSearchInput',
+      'change .js-sortBySelect': 'onChangeSortBy',
+    };
+  }
+
+  onFilterFreeShippingChange(e) {
+    this.filter.freeShipping = $(e.target).is(':checked');
+    this.renderListings(this.filteredCollection());
+  }
+
+  onShipsToSelectChange(e) {
+    if (this.$shipsToCheckbox.is(':checked')) {
+      this.setShipsToFilter($(e.target).val());
+    } else {
+      this.setShipsToFilter();
+    }
+  }
+
+  onShipsToCheckBoxChange(e) {
+    if ($(e.target).is(':checked')) {
+      this.setShipsToFilter(this.$shipsToSelect.val());
+    } else {
+      this.setShipsToFilter();
+    }
+  }
+
+  setShipsToFilter(val) {
+    if (val) {
+      this.filter.shipsTo = val;
+    } else {
+      delete this.filter.shipsTo;
+    }
+
+    this.renderListings(this.filteredCollection());
+  }
+
+  onChangeSortBy(e) {
+    this.filter.sortBy = $(e.target).val();
+    this.renderListings(this.storeListings.collection);
+  }
+
+  onUpdateCollection(cl, opts) {
+    if (opts.changes.added) {
+      opts.changes.added.forEach((md) => {
+        md.searchDescription = $('<div />').html(md.get('description'))
+          .text()
+          .toLocaleLowerCase();
+
+        md.searchTitle = $('<div />').html(md.get('title'))
+          .text()
+          .toLocaleLowerCase();
+      });
+    }
+  }
+
+  onKeyupSearchInput(e) {
+    // make sure they're not still typing
+    if (this.searchKeyUpTimer) {
+      clearTimeout(this.searchKeyUpTimer);
+    }
+
+    this.searchKeyUpTimer = setTimeout(() =>
+      (this.search($(e.target).val())), 150);
+  }
+
+  onRequest(cl, xhr) {
+    this.fetch = xhr;
```

Comment:
Yes, because this is a callback that I'm setting to be called (via backbone) on the collection's 'request' event:

`this.listenTo(this.collection, 'request', this.onRequest);`

I just happen to also need to execute that code path if the view is loaded with a request already in progress (in which case the request event already fired before we bound our callback to it). So... I'm just mimicking what Backbone sends to that callback.